PhysicsTools/NanoAOD/python/electrons_cff.py

```diff
@@ -298,7 +298,18 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
         jetIdx = Var("?hasUserCand('jet')?userCand('jet').key():-1", "int16", doc="index of the associated jet (-1 if none)"),
         photonIdx = Var("?overlaps('photons').size()>0?overlaps('photons')[0].key():-1", "int16", doc="index of the first associated photon (-1 if none)"),
         svIdx = Var("?hasUserCand('vertex')?userCand('vertex').key():-1", "int16", doc="index of matching secondary vertex"),
+        fbrem = Var("fbrem()",float,doc="Fraction of brem",precision=10),                 
+        rawEnergy = Var("superCluster.rawEnergy",float,doc="raw energy",precision=10),
+        rawPreshowerEnergy = Var("superCluster.preshowerEnergy",float,doc="energy deposited in preshower",precision=10),                 
+        ecalEnergy = Var("ecalEnergy()",float,doc="energy after ECAL-only regression applied",precision=10),
+        ecalEnergyError = Var("ecalEnergyError",float,doc="ecalEnergy error",precision=10),
         energyErr = Var("p4Error('P4_COMBINATION')",float,doc="energy error of the cluster-track combination",precision=6),
+        gsfTrkpMode = Var("gsfTrack().pMode()",float,doc="GSF track pMode",precision=10),
+        gsfTrkpModeErr = Var("abs(gsfTrack().qoverpModeError())*gsfTrack().pMode()*gsfTrack().pMode()",float,doc="GSF track pMode error",precision=8),
+        gsfTrketaMode = Var("gsfTrack().etaMode()",float,doc="GSF track etaMode",precision=10),
+        gsfTrkphiMode = Var("gsfTrack().phiMode()",float,doc="GSF track phiMode",precision=10),                 
+        isEcalDriven = Var("ecalDrivenSeed",bool,doc="is ECAL driven if true"),
+        isEB = Var("isEB",bool,doc="object in barrel if true"),                 
```

Comment:
please remove isEB, this can be infered from the eta.

Reply:
Hi @simonepigazzini  the decision of "isEB"  is from the seed crystal and detID information. It is beingused while deriving energy regression of electon by combing ecal and tracker information. We are trying to see if we can use this barrel/endcap definition from the supercluster eta of electron.

Reply:
yeah, I think the difference should be negligible. It would be actually very interesting to know how many times using supercluster.eta gives a different result that the seed rec hit position (I think this question has popped up few times in the past, maybe we already know the answer ....)

Reply:
I've processed 50k electrons (slimmedElectrons in Mini, no ID, no cutoffs of sort applied) and checked how many times **isEB** was giving different results wrt **etaSC** with a barrel cutoff at |eta| = 1.479. I found that only one electron had |eta| > 1.479 (slightly) when isEB == true. That makes it a 0.002% difference. My opinion is that we could go with etaSC and accept that there will be a difference of this order of magnitude wrt the regression done on the same events in MiniAOD.

Note that, if the barrel eta cutoff changes slightly (e.g., 1.4442, which is the number we typically use as the start of the calorimeter transition region), these numbers become dramatically different and the discrepancy is rather a 2.5% one.